patches/minecraft/net/minecraft/tileentity/TileEntityBeacon.java.patch

```diff
@@ -9,7 +9,27 @@
                      {
                          this.field_146015_k = false;
                          this.field_174909_f.clear();
-@@ -181,7 +181,7 @@
+                         break;
+                     }
+ 
+-                    tileentitybeacon$beamsegment.func_177262_a();
+-                    continue;
++                    // Allow any block to change the beacon beam colour
++                    float[] colour = iblockstate.func_177230_c().getBeaconColour(field_145850_b, iblockstate);
++                    if(colour == null || colour.length != 3){
++                        tileentitybeacon$beamsegment.func_177262_a();
++                        continue;
++                    }
++                    afloat = colour;
++                }else {
++                    afloat = EntitySheep.func_175513_a((EnumDyeColor) iblockstate.func_177229_b(BlockStainedGlassPane.field_176245_a));
+                 }
+-
+-                afloat = EntitySheep.func_175513_a((EnumDyeColor)iblockstate.func_177229_b(BlockStainedGlassPane.field_176245_a));
+             }
```

Comment:
It is because if I do not do that the color gets set to the value in the blockstate in the BlockStainedGlassPane, which will crash because it does not exist in most blocks.